Formula/lua-language-server.rb

```diff
@@ -0,0 +1,111 @@
+    # system "git", "submodule", "update", "--init", "--recursive"
+    chdir "3rd/luamake" do
+      system "compile/install.sh"
+    end
+    system "3rd/luamake/luamake", "rebuild"
+    bindir = if OS.mac?
+      "bin/macOS"
+    else
+      "bin/Linux"
+    end
+    (libexec/bindir).mkpath
+    (libexec/bindir).install "#{bindir}/lua-language-server"
+    (libexec/bindir).install "#{bindir}/main.lua"
+    libexec.install "main.lua", "debugger.lua"
+    libexec.install "locale", "meta", "script"
+    bin.write_exec_script libexec/bindir/"lua-language-server"
+    (libexec/"log").mkpath
+  end
+
+  test do
+    output = /Content-Length: \d+\r\n\r\n/
+
+    assert_match output, pipe_output("#{bin}/lua-language-server", 0)
+  end
+end
+
+__END__
+--- a/3rd/bee.lua/test/test_filesystem.lua
++++ b/3rd/bee.lua/test/test_filesystem.lua
+@@ -777,15 +777,15 @@
+ --    assertPathEquals(fs.dll_path(), fs.absolute(getdll()))
+ --end
+ 
+-function test_fs:test_appdata_path()
+-    if platform.OS == 'Windows' then
+-        assertPathEquals(fs.appdata_path(), os.getenv "LOCALAPPDATA")
+-    elseif platform.OS == 'Linux' then
+-        assertPathEquals(fs.appdata_path(), os.getenv "XDG_DATA_HOME" or (os.getenv "HOME" .. "/.local/share"))
+-    elseif platform.OS == 'macOS' then
+-        assertPathEquals(fs.appdata_path(), os.getenv "HOME" .. "/Library/Caches")
+-    end
+-end
++--function test_fs:test_appdata_path()
++--    if platform.OS == 'Windows' then
++--        assertPathEquals(fs.appdata_path(), os.getenv "LOCALAPPDATA")
++--    elseif platform.OS == 'Linux' then
++--        assertPathEquals(fs.appdata_path(), os.getenv "XDG_DATA_HOME" or (os.getenv "HOME" .. "/.local/share"))
++--    elseif platform.OS == 'macOS' then
++--        assertPathEquals(fs.appdata_path(), os.getenv "HOME" .. "/Library/Caches")
++--    end
++--end
+ 
+ 
+ function test_fs:test_filelock_1()
+
+--- a/3rd/luamake/3rd/bee.lua/test/test_filesystem.lua
++++ b/3rd/luamake/3rd/bee.lua/test/test_filesystem.lua
+@@ -777,15 +777,15 @@
+ --    assertPathEquals(fs.dll_path(), fs.absolute(getdll()))
+ --end
+ 
+-function test_fs:test_appdata_path()
+-    if platform.OS == 'Windows' then
+-        assertPathEquals(fs.appdata_path(), os.getenv "LOCALAPPDATA")
+-    elseif platform.OS == 'Linux' then
+-        assertPathEquals(fs.appdata_path(), os.getenv "XDG_DATA_HOME" or (os.getenv "HOME" .. "/.local/share"))
+-    elseif platform.OS == 'macOS' then
+-        assertPathEquals(fs.appdata_path(), os.getenv "HOME" .. "/Library/Caches")
+-    end
+-end
++--function test_fs:test_appdata_path()
++--    if platform.OS == 'Windows' then
++--        assertPathEquals(fs.appdata_path(), os.getenv "LOCALAPPDATA")
++--    elseif platform.OS == 'Linux' then
++--        assertPathEquals(fs.appdata_path(), os.getenv "XDG_DATA_HOME" or (os.getenv "HOME" .. "/.local/share"))
++--    elseif platform.OS == 'macOS' then
++--        assertPathEquals(fs.appdata_path(), os.getenv "HOME" .. "/Library/Caches")
```

Comment:
Do you know what `assertPathEquals` does when passed no arguments? I wonder if we can just `inreplace` the arguments here instead. Or perhaps mangle `test_fs:test_appdata_path` so that it doesn't get run during the tests.

Reply:
> Or perhaps mangle test_fs:test_appdata_path so that it doesn't get run during the tests.

Isn't that exactly what this patch is doing? It skips a test that is known to (necessary) fail if building with brew?

Reply:
(To be clear: I'm not modifying any part of the library itself, I'm just removing one test from a script that contains a bunch of tests that are hardcoded to run after each build.)

Reply:
Yes, but using an `inreplace` instead of a patch won't trigger the audit 🙂

Reply:
:D

Reply:
> Isn't that exactly what this patch is doing? It skips a test that is known to (necessary) fail if building with brew?

Yes, but I was suggesting something much simpler (and perhaps less brittle?) than a patch: mangle the string literals `test_fs:test_appdata_path` so that the test no longer runs them. The rest of the code can stay, we don't need to patch them out if they live in a function that is never called.

Reply:
Replacing the checks with `assertPathEquals("","")` builds locally. How would I make these changes using `inreplace`? It's a bit of a blunt tool...

Reply:
I mean, I can replace the two lines literally, but that seems just as brittle as a patch?

Reply:
Yea, it is just as brittle. That's why I was suggesting we just change the name of the function so that it never gets called.

Reply:
That won't work; the test is never called explicitly -- the build system just runs all tests that happen to be in that file (similar to `busted`).

Reply:
Right. Ok, that won't work then. How about
```ruby
home_subdir = OS.mac? ? "Library/Caches" : ".local/share"
inreplace buildpath.glob("**/3rd/bee.lua/test/test_filesystem.lua"), "fd.appdata_path()", "#{ENV["HOME"]}/#{home_subdir}"
```
?

Reply:
Maybe -- but then I'd rather just skip the test entirely. I've managed to patch in an early return; let's see if CI likes that ;)

Reply:
It does \o/